docs-src/java/how-to-use-a-custom-payload-converter-in-java.md

```diff
@@ -0,0 +1,67 @@
+
+The `PayloadConverter` is used to serialize/deserialize method parameters that need to be sent over the wire.
+You can create a custom implementation of the `PayloadConverter` for custom formats, for example:
+
+```java
+/** Payload converter specific to CloudEvents format */
+public class YourCustomPayloadConverter implements PayloadConverter {
+ //....
+  @Override
+  public String getEncodingType() {
+    return "json/plain";
+  }
+
+  @Override
+  public Optional<Payload> toData(Object value) throws DataConverterException {
+      //your convert-tologic here
+  }
+
+  @Override
+  public <T> T fromData(Payload content, Class<T> valueClass, Type valueType)
+      throws DataConverterException {
+    //your convert-from logic here.
+}
+```
+
+You can also use the following implementation classes provided in the Java SDK:
+
+- [ByteArrayPayloadConverter](https://www.javadoc.io/static/io.temporal/temporal-sdk/1.18.1/io/temporal/common/converter/ByteArrayPayloadConverter.html)
+- [GsonJsonPayloadConverter](https://www.javadoc.io/static/io.temporal/temporal-sdk/1.18.1/io/temporal/common/converter/GsonJsonPayloadConverter.html)
+- [JacksonJsonPayloadConverter](https://www.javadoc.io/static/io.temporal/temporal-sdk/1.18.1/io/temporal/common/converter/JacksonJsonPayloadConverter.html)
+- [NullPayloadConverter](https://www.javadoc.io/static/io.temporal/temporal-sdk/1.18.1/io/temporal/common/converter/NullPayloadConverter.html)
+- [ProtobufJsonPayloadConverter](https://www.javadoc.io/static/io.temporal/temporal-sdk/1.18.1/io/temporal/common/converter/ProtobufJsonPayloadConverter.html)
+- [ProtobufPayloadConverter](https://www.javadoc.io/static/io.temporal/temporal-sdk/1.18.1/io/temporal/common/converter/ProtobufPayloadConverter.html)
+
+For example, to create a custom `JacksonJsonPayloadConverter`, use:
+
+```java
+private static JacksonJsonPayloadConverter yourCustomJacksonJsonPayloadConverter() {
+  ObjectMapper objectMapper = new ObjectMapper();
+  // your custom conversion logic
+  return new JacksonJsonPayloadConverter(objectMapper);
+}
+```
+
+To set your custom `PayloadConverter`, use [`withPayloadConverterOverrides`](https://www.javadoc.io/doc/io.temporal/temporal-sdk/latest/io/temporal/common/converter/DefaultDataConverter.html#withPayloadConverterOverrides(io.temporal.common.converter.PayloadConverter...)) with a new instance of the `DefaultDataConverter` to override default behavior with the custom `PayloadConverter` in your `WorkflowClient` options that you use in your Worker process and to start your Workflow Executions.
+
+The following example shows how to set a custom `YourCustomPayloadConverter` Payload Converter.
+
+```java
+DefaultDataConverter ddc =
+        DefaultDataConverter.newDefaultInstance()
+            .withPayloadConverterOverrides(new YourCustomPayloadConverter());
```

Comment:
Probably would be good to mention that encodingType will be used to define what to we "overriding"